### ISSUE TITLE
"Address already in use" even if no containers connected

### ISSUE THREAD

**Issue by alehaa** [docker/libnetwork#2425]
I regularly get the "Address already in use" error when recreating an existing container (by `docker-compose`), even if the container being recreated is the only member of this macVLAN network (as pointed out by `docker network inspect`).

Although the network shouldn't have any member left at the moment (as the container is deleted / recreated) and `docker network inspect` confirms this status, the underlaying IPAM database seems to have a state not matching to this output. However, I can't find a way to tell it to prune all records of addresses in use.

**Comment by gun1x** [comment]
This might be a duplicate of https://github.com/docker/libnetwork/issues/1447

Is the IP you are trying to use magically reserved for the gateway, even though you didn't specify a gateway?

**Comment by alehaa** [comment]
No, my gateway is .1, but problems happen with e.g. .70. I need to delete the database and reconnect all containers with a new network with same specs in that case.
